src/tools/miri/tests/pass-dep/libc/libc-time.rs

```diff
@@ -60,6 +73,82 @@ fn test_posix_gettimeofday() {
     assert_eq!(is_error, -1);
 }
 
+fn test_nanosleep() {
+    let start_test_sleep = Instant::now();
+    let duration_zero = libc::timespec { tv_sec: 0, tv_nsec: 0 };
+    let remainder = ptr::null_mut::<libc::timespec>();
+    let is_error = unsafe { libc::nanosleep(&duration_zero, remainder) };
+    assert_eq!(is_error, 0);
+    assert!(start_test_sleep.elapsed() < Duration::from_millis(10));
+
+    let start_test_sleep = Instant::now();
+    let duration_100_millis = libc::timespec { tv_sec: 0, tv_nsec: 1_000_000_000 / 10 };
+    let remainder = ptr::null_mut::<libc::timespec>();
+    let is_error = unsafe { libc::nanosleep(&duration_100_millis, remainder) };
+    assert_eq!(is_error, 0);
+    assert!(start_test_sleep.elapsed() > Duration::from_millis(100));
+}
+
+/// Helper function to get the current time for testing relative sleeps
+fn timespec_now(clock: libc::clockid_t) -> libc::timespec {
+    let mut timespec = mem::MaybeUninit::<libc::timespec>::uninit();
+    let is_error = unsafe { libc::clock_gettime(clock, timespec.as_mut_ptr()) };
+    assert_eq!(is_error, 0);
+    unsafe { timespec.assume_init() }
+}
+
+fn test_clock_nanosleep_absolute() {
```

Comment:
These functions also need the `cfg` to not be built on macOS.

You can test this locally even if you are not on macOS:
```
./x test miri --target aarch64-apple-darwin  -- time
```

Reply:
> You can test this locally even if you are not on macOS:

that's super useful thanks!